consensus/misc/eip4844.go

```diff
@@ -45,16 +47,30 @@ func CalcExcessDataGas(parentExcessDataGas *big.Int, newBlobs int) *big.Int {
 
 // FakeExponential approximates factor * e ** (num / denom) using a taylor expansion
 // as described in the EIP-4844 spec.
-func FakeExponential(factor, num, denom *big.Int) *big.Int {
-	output := new(big.Int)
-	numAccum := new(big.Int).Mul(factor, denom)
-	for i := 1; numAccum.Sign() > 0; i++ {
-		output.Add(output, numAccum)
-		numAccum.Mul(numAccum, num)
-		iBig := big.NewInt(int64(i))
-		numAccum.Div(numAccum, iBig.Mul(iBig, denom))
+func FakeExponential(factor, denom *uint256.Int, edg *big.Int) (*uint256.Int, error) {
+	numerator, overflow := uint256.FromBig(edg)
+	if overflow {
+		return nil, fmt.Errorf("FakeExponential: overflow converting excessDataGas: %v", edg)
+	}
+	output := uint256.NewInt(0)
+	numeratorAccum := new(uint256.Int)
+	_, overflow = numeratorAccum.MulOverflow(factor, denom)
+	if overflow {
+		return nil, fmt.Errorf("FakeExponential: overflow in MulOverflow(factor=%v, denom=%v)", factor, denom)
+	}
+	divisor := new(uint256.Int)
+	for i := 1; numeratorAccum.Sign() > 0; i++ {
+		output.Add(output, numeratorAccum)
```

Comment:
I guess to be safe we should use AddOverflow here as well. Kind of a pain but better safe than sorry.